yb-voyager/src/srcdb/postgres.go

```diff
@@ -945,3 +959,534 @@ func (pg *PostgreSQL) ValidateTablesReadyForLiveMigration(tableList []sqlname.Na
 	}
 	return nil
 }
+
+// =============================== Guardrails ===============================
+
+func (pg *PostgreSQL) CheckSourceDBVersion() error {
+	utils.PrintAndLog("checking postgres version")
+	pgVersion := pg.GetVersion()
+	if pgVersion == "" {
+		return fmt.Errorf("failed to get source database version")
+	}
+	supportedVersionRange := fmt.Sprintf("from %s to %s", MIN_SUPPORTED_PG_VERSION_OFFLINE, MAX_SUPPORTED_PG_VERSION)
+
+	if version.CompareSimple(pgVersion, MAX_SUPPORTED_PG_VERSION) > 0 || version.CompareSimple(pgVersion, MIN_SUPPORTED_PG_VERSION_OFFLINE) < 0 {
+		return fmt.Errorf("source database version %s is not supported. Supported versions are %s", pgVersion, supportedVersionRange)
+	}
+	if version.CompareSimple(pgVersion, MIN_SUPPORTED_PG_VERSION_LIVE) < 0 {
+		utils.PrintAndLog(color.RedString("Warning: Source database version %s is not supported for live migration. Supported versions are %s", pgVersion, supportedVersionRange))
+	}
+
+	return nil
+}
+
+// GetMissingExportSchemaPermissions checks for missing permissions required for exporting schema in a PostgreSQL database.
+// It verifies if schemas have USAGE permission
+// and if tables in the provided schemas + pg_catalog + infomation_schema have SELECT permission.
+// Returns:
+//   - []string: A slice of strings describing the missing permissions, if any.
+//   - error: An error if any issues occur during the permission checks.
```

Comment:
minor nit: Using multi line comment(/**/) in such cases can enhance readability.